docs/advanced_documentation/c-api.md

```diff
@@ -69,45 +69,86 @@ In the `PGM_calculate` function you need to pass a pointer to `PGM_Options`.
 In this way, we can ensure the API backwards compatibility.
 If we add a new option, it will get a default value in the `PGM_create_options` function.
 
-## Buffer and Attributes
+## Buffers and attributes
+
+The biggest challenge in the design of the C API is the handling of input/output/update data communication.
+To this end, data is communicated using buffers, which are contiguous blocks of memory that represent data in a given format.
+For compatibility reasons, that format is dictated by the C API using the `PGM_meta_*` functions and `PGM_def_*` dataset definitions.
 
-The biggest challenge in the design of C API is the handling of input/output/update data buffers.
 We define the following concepts in the data hierarchy:
 
 * Dataset: a collection of data buffers for a given purpose. 
-At this moment, we have four dataset types: `input`, `update`, `sym_output`, `asym_output`.
-* Component: a homogeneous data buffer for a component in our [data model](../user_manual/components.md), e.g., `node`.
+  At this moment, we have four dataset types: `input`, `update`, `sym_output`, `asym_output`.
+* Component: the representation of attributes for the same component in our [data model](../user_manual/components.md), e.g., `node`.
```

Comment:
I don't think I like the wording here, it substracts importance to the Component itself and doesn't tie well with the Dataset definition. What about something like "A data buffer for a physical  grid components, characterized by its attributes, in our..."